File: pilhas-filas/python/remove_duplicates.py
def remove_duplicates(text):
    stack = []
    for char in text:
        if stack and stack[-1] == char:
            stack.pop()
        else:
            stack.append(char)
    return ''.join(stack)


print(remove_duplicates("abbaca"))
print(remove_duplicates("azxxzy"))

File: pilhas-filas/javascript/remove-duplicates.js
function removeDuplicates(text) {

    let stack = [];
    for (let char of text) {
        if (stack.length != 0 && stack[stack.length - 1] == char) {
            stack.pop();
        }
        else {
            stack.push(char);
        }
    }

    return stack.join("");
}

console.log(removeDuplicates("abbaca"));
console.log(removeDuplicates("azxxzy"));
